### ISSUE TITLE
Integration with @storybook/addon-info or custom layouts.

### ISSUE THREAD

**Issue by chemoish** [tuchk4/storybook-readme#82]
This may be a dumb question, but would there be any consideration to provide a HOC that would be more general and just transform the markdown itself?

These are already great:
- Panel (`withReadme`)
- Story (`doc`)
- Wrapper (`withDocs` with preview component)
- Wrapper defined by README (`withDocs` with  comment)

I was thinking something like this?

```jsx
import { withInfo } from '@storybook/addon-info';
import { withMarkdown } from 'storybook-readme';

import INTRO from './INTRO.md';
import OUTRO from './OUTRO.md';

storiesOf('Introduction', module)
  .add('Test', withInfo()(
    <div>
      {withMarkdown(INTRO)}

      <div>Hi</div>

      {withMarkdown(OUTRO)}
    </div>
  )
;
```

It _might_ be able to replace the wrapper cases, assuming people want to split up their markdown files.

Thoughts?

---

__Context__

It looks like adding `withInfo` is placed within the `PreviewComponent`, defaulted to https://github.com/tuchk4/storybook-readme/blob/master/packages/storybook-readme/src/env/react/components/StoryPreview.js.

![screen shot 2018-08-07 at 6 28 23 pm](https://user-images.githubusercontent.com/549319/43811586-135fc404-9a72-11e8-9c02-e3bbc1def0d2.png)

**Comment by tuchk4** [comment]
I will provide this component in next release

**Comment by tuchk4** [comment]
@chemoish take a look at this example https://tuchk4.github.io/storybook-readme/?selectedKind=Marked&selectedStory=Marked1&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybooks%2Fstorybook-addon-knobs


Here is docs https://github.com/tuchk4/storybook-readme#custom-layout
Now feature available in `4.0.0-beta1`

**Comment by chemoish** [comment]
@tuchk4 Looks great!

Minor overall feedback:

1. Would be great to link your storybook in the repository's description (little bit hard to find in the middle of the `README`).
1. Would be nice if storybook followed the same hierarchy as the `README`, slight challenge to map storybook to `README`.
1. Possibly update `README`, or story, to match (e.g. Custom Layout > Marked)—Currently its "Custom layout" in the `README` and Marked > Marked1 in the story.
1. I understand `README` is a different than the story, but there could be a minor correction in the `README`.

    ```
    -<Button label="Button before docs" />
    +<Button label="Button before intro" />
    ```

Many of these things are very nit, but thanks again for all your hard work!

**Comment by tuchk4** [comment]
- [x] Would be great to link your storybook in the repository's description (little bit hard to find in the middle of the README).

- [ ] Would be nice if storybook followed the same hierarchy as the README, slight challenge to map storybook to README.

- [x]  Possibly update README, or story, to match (e.g. Custom Layout > Marked)—Currently its "Custom layout" in the README and Marked > Marked1 in the story.

- [x]  I understand README is a different than the story, but there could be a minor correction in the README.

**Comment by tuchk4** [comment]
released in `4.0.0`
